addons/l10n_nl/migrations/12.0.3.0/pre-migration.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+
+def merge_res_country_states(cr):
+    openupgrade.logged_query(
+        cr, """
+        INSERT INTO ir_model_data (module, name, model, res_id, noupdate)
+          SELECT 'l10n_nl',
+                 CONCAT('state_nl_', LOWER(s.code)),
+                 'res.country.state',
+                 s.id,
+                 false
+          FROM res_country_state s
+              JOIN res_country c ON s.country_id = c.id
+              WHERE c.code LIKE 'NL'
```

Comment:
I think you should limit possible states here (as there can be manual entries created even without codes). Something like:
```
AND s.code IN ('bq1', 'bq2', 'bq3', 'dr'...)
```